extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/graal/BsonSubstitutions.java

```diff
@@ -0,0 +1,106 @@
+package io.quarkus.mongodb.runtime.graal;
+
+import java.text.ParsePosition;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
+
+import org.bson.json.JsonParseException;
+import org.bson.json.JsonReader;
+
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+
+import io.quarkus.arc.impl.Reflections;
+
+public final class BsonSubstitutions {
+}
+
+@TargetClass(JsonReader.class)
+final class JsonReaderSubstitution {
+    @Substitute
+    private long visitISODateTimeConstructor() {
```

Comment:
As this is a private method, we run the risk that the driver may in the future refactor the JsonReader class in a way that makes this substitution incorrect;  for example, the method might be removed entirely.  We've had this experience before years ago when APM like App Dynamics vendors started patching our drivers using a similar technique, and then would break when the internal structure of the drivers changed.